### DISCUSSION TITLE
How can I display video in Avalonia UI (but in a way that works on all platforms)?

### DISCUSSION THREAD

**Discussion by al-zakaria** [AvaloniaUI/Avalonia#17801] [Q&A]
I know this question has been asked a lot (about displaying video), but I couldn't find a solution for the browser (maybe I didn't search well or didn't read the docs thoroughly, but I still haven't found anything). Yeah, for browser-based stuff, it's better to use JS or Blazor, but I specifically need Avalonia because it's what I'm most familiar with compared to Blazor and especially JS.

**Comment by maxkatz6** [comment]
Avalonia doesn't have a built-in MediaPlayer control yet (it's going to change with [new tools we are currently building](https://github.com/AvaloniaUI/Avalonia/discussions/16997)).

There are several third-party options:
- https://github.com/mysteryx93/MediaPlayerUI.NET
- https://github.com/radiolondra/YAMP2
- https://github.com/donandren/vlcsharpavalonia (outdated)

And as another option, you can use native control embedding, via NativeControlHost. But it would require writing a wrapper per each platform you need to target.

**Reply by thevortexcloud** [reply]
The up to date VLC integration for Avalonia is hosted on the official VLC gitlab.

https://code.videolan.org/videolan/LibVLCSharp/-/tree/master/samples/LibVLCSharp.Avalonia.Sample

> And as another option, you can use native control embedding, via NativeControlHost. But it would require writing a wrapper per each platform you need to target.

There is a sample here on how to do that for a HTML element (at which point you can just use a standard video tag):

https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Browser/EmbedSample.Browser.cs

**Reply by al-zakaria** [reply]
And when will the update be approximately?

I also read that all the innovations that you have left a link to will be paid for. Is Avalonia becoming paid? And some of the functionality will not be released for remote users? What should I do in this case?

**Reply by thevortexcloud** [reply]
Which update? The official media player from Avalonia? It's listed as a phase 1 release which is "soon". However it's not so much an update for Avalonia but a separate paid commercial product with different licensing. If you want something for free then your best bet is to either make your own or use an existing implementation. There is quite a few people have worked on. EG most recently this one showed up: https://github.com/AvaloniaUI/Avalonia/discussions/17740

The core Avalonia is at least at this stage, staying FOSS MIT licensed. The paid controls are for more niche use cases and to also give some extra revenue to the company that makes the FOSS Avalonia.

**Reply by maxkatz6** [reply]
> Is Avalonia becoming paid?

No, it's not. Avalonia is free and open-source project, we are not going to change that.

But Accelerate is a new set of products, built with and for Avalonia. These won't be free.

**Reply by al-zakaria** [reply]
And what will be the prices?
